packages/block-editor/src/components/block-list/use-block-props/use-zoom-out-mode-exit.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useRefEffect } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../../store';
+import { unlock } from '../../../lock-unlock';
+
+/**
+ * Allows Zoom Out mode to be exited by double clicking in the selected block.
+ *
+ * @param {string} clientId Block client ID.
+ */
+export function useZoomOutModeExit( clientId ) {
+	const { isZoomOutMode, isBlockSelected } = unlock(
+		useSelect( blockEditorStore )
+	);
+	const { __unstableSetEditorMode, selectBlock } = unlock(
+		useDispatch( blockEditorStore )
+	);
+
```

Comment:
Zoom out mode is fairly rare right now. Rather than adding a double click to all blocks, could we early return here if we're not in zoom out? That way, we only add these listeners when it's useful?